### DISCUSSION TITLE
IConnectionFactory is missing HostName

### DISCUSSION THREAD

**Discussion by astaerk** [rabbitmq/rabbitmq-dotnet-client#1605] [General]
I was trying to inject an IConnectionFactory and then use it to create a connection.
But the HostName  property of the ConnectionFactory is missing on the IConnectionFactory interface.

### Reproduction steps

This code

```
IConnectionFactory connectionFactory = new ConnectionFactory();
connectionFactory.HostName = "hostName";
```

leads to the compilation error

```
CS1061	'IConnectionFactory' does not contain a definition for 'HostName' and no accessible extension method 'HostName' accepting a first argument of type 'IConnectionFactory' could be found (are you missing a using directive or an assembly reference?)
```

### Expected behavior

The property HostName of the ConnectionFactory is available on the IConnectionFactory interface.
Therefore the above code compiles without any error.

**Comment by michaelklishin** [comment]
We cannot suggest much since you haven't shared what version you use. We do not guess in this community.

In `main` (future 7.0), this is the case because [individual connection methods](https://github.com/rabbitmq/rabbitmq-dotnet-client/blob/bf15067058f7e234f0b4961c13d006ef9fc9ec00/projects/RabbitMQ.Client/client/api/IConnectionFactory.cs#L126) accept a list of endpoints to use. This is a reasonable API design choice.

**Comment by astaerk** [comment]
I'm using version 6.8.1, sorry for not including that information.

Actually I don't even need a suggestion what to do, because in the end I had to use the CreateConnection overload with IEnumerable<AmqpTcpEndpoint> for totally different reasons anyway.

The HostName property is used in many examples, although not on the interface but on the class. So for the sake of correctness I wanted to let you know about the missing property. Should be an easy fix after all.

But feel free to reject this issue if you don't think it is relevant.

**Reply by lukebakken** [reply]
If you must use `IConnectionFactory`, use the `Uri` property.

Note that version 7 will ship soon and it's too late to change these interfaces.

Version 8 will probably do away with most or all interfaces in this library, and use .NET types as much as possible (like `System.Net.Endpoint`)

**Reply by astaerk** [reply]
Thank you for that information. I'll keep that in mind.
